packages/nextjs/components/ExampleUi/ContractInteraction/PurposeSetter.tsx

```diff
@@ -1,6 +1,23 @@
+import { useState } from "react";
 import { ArrowSmallRightIcon } from "@heroicons/react/24/outline";
+import { useScaffoldContractWrite, useTransactor } from "~~/hooks/scaffold-eth";
+import { getParsedEthersError } from "~~/components/scaffold-eth/Contract/utilsContract";
+import { toast } from "~~/utils/scaffold-eth";
 
 export const PurposeSetter = () => {
+  const [newPurpose, setNewPurpose] = useState("");
+
+  // Could we move useTransactor inside useScaffoldContractWrite?
+  const writeTxn = useTransactor();
+  const { writeAsync, isLoading } = useScaffoldContractWrite(
+    "YourContract",
+    "setPurpose",
+    [newPurpose],
+    // Since this hook is going to be "high-level" maybe we should accept the value in ETH (not wei)?
+    // Passing "0.01" seems more high-level-dx-friendly.
+    "10000000000000000",
```

Comment:
Since this hook is going to be "high-level" maybe we should accept the value in ETH (not wei)?

Passing "0.01" seems more high-level-dx-friendly.

Thoughts?